### ISSUE TITLE
Feature Request: support close inside function call

### ISSUE THREAD

**Issue by dandandy** [ryanrolds/sqlclosecheck#7]
I would like to be able call a function to close my sqlx.Stmt structs to reduce dublicate code using a function like:
```
func closer(aStruct io.Closer) {
	err := aStruct.Close()
	if err != nil {
		logrus.Warnf("error closing sqlx statement %v", err)
	}
}
```

````
statement, err := db.Preparex(findUserOptions)
defer closer(statement)
````

But the version of sqlclosecheck in golangci-lint v1.30 will throw an error because it thinks I'm not closing the struct

`
pkg/database/client.go:174:34: Rows/Stmt was not closed (sqlclosecheck)
                statement, err := c.db.Preparex(findUserOptions)
`

**Comment by ryanrolds** [comment]
I will see what I can do.

**Comment by vovanec** [comment]
Is there any estimate when this can be fixed?

**Comment by urgas9** [comment]
big +1 this would help us too, we are wrapping the Close function in a helper function with signature `db.CloseResource(ctx, resource)`

**Comment by ocket8888** [comment]
This would be nice for some code I maintain, because I, too, have a wrapper function - `log.Close(resource)` - and my choices are to either do:
```go
rows, err := tx.Query(query)
if err != nil {
    return err
}
defer rows.Close()
```
which causes a lint issue with `errcheck` because the error return isn't being checked, or:
```go
rows, err := tx.Query(query)
if err != nil {
    return err
}
defer log.Close(rows)
```
which causes this lint "issue" to appear.

**Comment by ryanrolds** [comment]
I will spend some time on this project over the next few weeks. I will take a look at this.

**Comment by githoober** [comment]
@ryanrolds are you still planning on fixing it? Do you need any help?

**Comment by ryanrolds** [comment]
Yes. I'm currently out of spoons. Also, I'm fairly new to writting/maintaining linters and this linter is pushing against the boundaries of my understanding of the AST/SSA.

Contributions are very welcome.
